site/src/Librecores/ProjectRepoBundle/RepoCrawler/GithubRepoCrawler.php

```diff
@@ -0,0 +1,174 @@
+
+    /**
+     * @var Client
+     */
+    private $githubClient;
+
+    /**
+     * @var GithubApiService
+     */
+    private $githubApi;
+
+    /**
+     * @var string
+     */
+    private $githubUser;
+
+    /**
+     * @var string
+     */
+    private $githubRepoName;
+
+    public function __construct(SourceRepo $repo, MarkupToHtmlConverter $markupConverter,
+                                ProcessCreator $processCreator, ObjectManager $manager,
+                                LoggerInterface $logger, GithubApiService $ghApi)
+    {
+        parent::__construct($repo, $markupConverter, $processCreator, $manager, $logger);
+        $this->githubApi = $ghApi;
+        preg_match(static::GH_REGEX, $this->repo->getUrl(), $matches);
+        $this->githubUser = $matches[1];
+        $this->githubRepoName = $matches[2];
+    }
+
+    /**
+     * Check whether the given URL is a valid Github repository URL
+     * @param $repoUrl
+     * @return bool
+     */
+    public static function isGithubRepoUrl($repoUrl)
+    {
+        return preg_match(static::GH_REGEX, $repoUrl);
+    }
+
+    /**
+     * {@inheritDoc}
+     * @see RepoCrawler::isValidRepoType()
+     */
+    public function isValidRepoType(): bool
+    {
+        return static::isGithubRepoUrl($this->repo->getUrl());
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function updateSourceRepo()
+    {
+        $this->updateGithubMetrics();
+        parent::updateSourceRepo();
+    }
+
+    private function getGithubClient()
+    {
+
+        if (null === $this->githubClient) {
+            $user = $this->repo->getProject()->getParentUser();
+
+            if (null === $user) {
+                $user = $this->repo->getProject()->getParentOrganization()->getCreator();
+            }
+
+            $this->githubClient = $this->githubApi->getAuthenticatedClientForUser($user);
+
+            // TODO: Fallback to personal token if an account is still not available
+        }
+
+        return $this->githubClient;
+    }
+
+    /**
+     * Extract and update Github specific metrics
+     */
+    private function updateGithubMetrics()
+    {
+
+        $query =
+            <<<'QUERY'
+query getRepoData (
+  $owner: String!
+  $repository: String!
+ ) {
+  repository(owner: $owner, name: $repository) { 
+    forks {
+      totalCount
+    }
+    hasIssuesEnabled
+    updatedAt
+    pushedAt
+    issues(states:OPEN) {
+      totalCount
+    }
+    pullRequests(states:OPEN) {
+      totalCount
+    }
+    stargazers {
+      totalCount
+    }
+    watchers {
+      totalCount
+    }
+  }
+}
+QUERY;
+
+        $variables = [
+            'owner' => $this->githubUser,
+            'repository' => $this->githubRepoName,
+        ];
+
+        $client = $this->getGithubClient();
+
+        if (null === $client) {
+            // we cannot extract any data from github in this case
+            $this->logger->error('Unable to fetch data from Github: No access token');
```

Comment:
It makes a presumption that this is an only cause of null client. It may change in the future. In languages like Java I would rather recommend throwing exceptions, not sure about PHP